src/transpiler/babel.ts

```diff
@@ -52,7 +52,7 @@ export function getForcedBabelOptions({cwd}: GetForcedBabelOptionsOptions): Tran
 		sourceType: "module",
 		plugins: [
 			// Needed to make babel understand dynamic imports
-			resolveModule("@babel/plugin-syntax-dynamic-import")
+			resolveModule("@babel/plugin-syntax-dynamic-import", resolveModule("@babel/preset-env"))
```

Comment:
Shouldn't we just declare a dependency on `@babel/plugin-syntax-dynamic-import` and resolve it directly? We are now again assuming some other packages will bring in what we depend on, instead of just declaring our own dependencies properly, right?

Reply:
Ideally `@babel/plugin-syntax-dynamic-import` should be added as a peer dependency. I was mainly trying to avoid introducing a breaking change.

Yes it relies on `@babel/preset-env` having a dependency on `@babel/plugin-syntax-dynamic-import`. This change it least removes the reliance on it being hoisted to the top-level under `node_modules`.

Reply:
Two things:

1. This particular change will cause `babel/preset-env` to be loaded as a  _forced_ babel option. Also, as far as I'm aware it won't actually work, since presets must be provided to babel via the `presets` property instead. But [as described here](https://github.com/wessberg/rollup-plugin-ts#babelpreset-env-behavior-and-how-to-opt-out), it should be possible to use Babel without `@babel/preset-env` being applied. All we really want here is for Babel to at the very least support dynamic imports, considering it is such a fundamental part of bundling together an application or library, and we'll need Babel to understand that Syntax for Rollup to do its thing.

2. To my knowledge, this won't in any way affect #185, as it will still end up attempting to resolve `@babel/plugin-syntax-dynamic-import` without there being an explicit peer dependency on it.

Historically, the reason why there were no declared dependencies on these babel plugins is that there were a ton of them, and I didn't find the user experience nice at all to ask users to install a gazillion packages themselves just to get started, especially when they're all covered by `@babel/preset-env`.

But now, as we no longer forcefully load the other syntax plugins like object-rest-spread, etc, I think I'm in favor of adding an explicit dependency on `@babel/plugin-syntax-dynamic-import` as a peer dependency and making it a breaking change, as @simonihmig also suggested.

Reply:
I think you misread. I am not adding `@babel/preset-env` as a plugin, I am passing `resolveModule("@babel/preset-env")` as a second argument to `resolveModule`. This is related to the other change you are asking about.

Currently the dependency tree of a project using `rollup-plugin-ts` looks like this

```
<project>
├ rollup-plugin-ts -- (peer) --> @babel/preset-env
└ @babel/present-env
    └ @babel/plugin-syntax-dynamic-import
```

With a strict resolution strategy like PNP, code inside `rollup-plugin-ts` cannot import `@babel/plugin-syntax-dynamic-import` since it is not a dependency nor a peer dependency. However, `rollup-plugin-ts` can resolve `@babel/preset-env`, and `@babel/preset-env` can resolve `@babel/plugin-syntax-dynamic-import`.

To do this, we need use `createRequire(import.meta.url).resolve('@babel/preset-env')`, which returns the path to a module inside `@babel/preset-env`, then `createRequire(path).resolve` to resolve *on behalf of* `@babel/preset-env`. Thus the changes here and in `path-utils.ts`.

----

But this is all moot if you are open to making this a breaking change.

Reply:
Aha, I see. You're right that I misread that. That _does_ seem really smart. While I realize there's a slight risk that `@babel/preset-env` will eventually drop its own dependency on `@babel/plugin-syntax-dynamic-import`, it does feel very unlikely. Given that, let's proceed with the change you've made here. Good job!

Reply:
> there's a slight risk that @babel/preset-env will eventually drop its own dependency on @babel/plugin-syntax-dynamic-import

Exactly. I'll proceed with my original changes but I have added a TODO comment to actually add a peer dep on `@babel/plugin-syntax-dynamic-import` if and when preparing the next major.